src/nunit.analyzers/ConstraintUsage/EqualToConstraintUsageAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+            id: AnalyzerIdentifiers.IsEqualToConstraintUsage,
+            title: EqualToConstraintUsageConstants.IsEqualToTitle,
+            messageFormat: EqualToConstraintUsageConstants.IsEqualToMessage,
+            category: Categories.Assertion,
+            defaultSeverity: DiagnosticSeverity.Info,
+            description: EqualToConstraintUsageConstants.IsEqualToDescription);
+
+        private static readonly DiagnosticDescriptor notEqualToDescriptor = DiagnosticDescriptorCreator.Create(
+            id: AnalyzerIdentifiers.IsNotEqualToConstraintUsage,
+            title: EqualToConstraintUsageConstants.IsNotEqualToTitle,
+            messageFormat: EqualToConstraintUsageConstants.IsNotEqualToMessage,
+            category: Categories.Assertion,
+            defaultSeverity: DiagnosticSeverity.Info,
+            description: EqualToConstraintUsageConstants.IsNotEqualToDescription);
+
+        private static readonly string[] SupportedPositiveAssertMethods = new[] {
+            NunitFrameworkConstants.NameOfAssertThat,
+            NunitFrameworkConstants.NameOfAssertTrue,
+            NunitFrameworkConstants.NameOfAssertIsTrue
+        };
+
+        private static readonly string[] SupportedNegativeAssertMethods = new[] {
+            NunitFrameworkConstants.NameOfAssertFalse,
+            NunitFrameworkConstants.NameOfAssertIsFalse
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            equalToDescriptor, notEqualToDescriptor);
+
+
+        protected override void AnalyzeAssertInvocation(SyntaxNodeAnalysisContext context, InvocationExpressionSyntax assertExpression, IMethodSymbol methodSymbol)
+        {
+            var negated = false;
+
+            if (SupportedNegativeAssertMethods.Contains(methodSymbol.Name))
+            {
+                negated = !negated;
+            }
+            else if (!SupportedPositiveAssertMethods.Contains(methodSymbol.Name))
+            {
+                return;
+            }
+
+            var constraint = assertExpression.GetArgumentExpression(methodSymbol, NunitFrameworkConstants.NameOfExpressionParameter);
+
+            // Constraint should be either absent, or Is.True, or Is.False
+            if (constraint != null)
+            {
+                if (!(constraint is MemberAccessExpressionSyntax memberAccess
+                       && memberAccess.Expression is IdentifierNameSyntax identifierSyntax
+                       && identifierSyntax.Identifier.Text == NunitFrameworkConstants.NameOfIs
+                       && (memberAccess.Name.Identifier.Text == NunitFrameworkConstants.NameOfIsTrue
+                           || memberAccess.Name.Identifier.Text == NunitFrameworkConstants.NameOfIsFalse)))
+                {
+                    return;
+                }
+                else if (memberAccess.Name.Identifier.Text == NunitFrameworkConstants.NameOfIsFalse)
```

Comment:
I think that `memberAccess` can be null.

Reply:
Not sure. Why? It is checked in 'if' above